### PR TITLE
2.5.3 fixes

### LINKED ISSUE #1260: Crash Report: IllegalArgumentException FileUtils.contentQuery (FileUtils.java:159)
**Describe the bug**
```
java.lang.RuntimeException: 

 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2756)

 
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2825)

 
  at android.app.ActivityThread.-wrap12 (ActivityThread.java)

 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1557)

 
  at android.os.Handler.dispatchMessage (Handler.java:110)

 
  at android.os.Looper.loop (Looper.java:203)

 
  at android.app.ActivityThread.main (ActivityThread.java:6339)

 
  at java.lang.reflect.Method.invoke (Native Method)

 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1084)

 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:945)
Caused by: java.lang.IllegalArgumentException: 

 
  at android.database.AbstractCursor.getColumnIndexOrThrow (AbstractCursor.java:333)

 
  at android.database.CursorWrapper.getColumnIndexOrThrow (CursorWrapper.java:87)

 
  at org.kiwix.kiwixmobile.utils.files.FileUtils.contentQuery (FileUtils.java:159)

 
  at org.kiwix.kiwixmobile.utils.files.FileUtils.getLocalFilePathByUri (FileUtils.java:144)

 
  at org.kiwix.kiwixmobile.KiwixMobileActivity.manageExternalLaunchAndRestoringViewState (KiwixMobileActivity.java:1771)

 
  at org.kiwix.kiwixmobile.KiwixMobileActivity.onCreate (KiwixMobileActivity.java:462)

 
  at android.app.Activity.performCreate (Activity.java:6689)

 
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1140)

 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2709)
```

### LINKED ISSUE #1261: Crash Report: IndexOutOfBoundsException KiwixMobileActivity.onResume (KiwixMobileActivity.java:1263)
**Describe the bug**
```
java.lang.RuntimeException: 

 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4290)

 
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4330)

 
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:51)

 
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:145)

 
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:70)

 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2123)

 
  at android.os.Handler.dispatchMessage (Handler.java:109)

 
  at android.os.Looper.loop (Looper.java:207)

 
  at android.app.ActivityThread.main (ActivityThread.java:7470)

 
  at java.lang.reflect.Method.invoke (Native Method)

 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:524)

 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:958)
Caused by: java.lang.IndexOutOfBoundsException: 

 
  at java.util.ArrayList.get (ArrayList.java:437)

 
  at org.kiwix.kiwixmobile.KiwixMobileActivity.onResume (KiwixMobileActivity.java:1263)

 
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1456)

 
  at android.app.Activity.performResume (Activity.java:7592)

 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4272)
```

### PR DESCRIPTION
**Pull request by abdulwd**
Fixes #1260, #1261

### CODE REVIEW

**Comment by kelson42** [comment]
@macgills Milestone 3.0 backlog is slowly reducing but I still expect a few weeks before we can publicly release it. I have been told by @julianharty that, following his computation, these both fixes by @abdulwd should allow us to reduce our app crash rate by 0.4 -> 0.6 points which makes around 30% improvement. To me, and considering that we still have a crash rate twice as big as what is tolerable by users/Google, it would be worth a new small maintenance release.

File: app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.java [hunk 1]
```diff
@@ -25,22 +25,18 @@
...
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
+import org.kiwix.kiwixmobile.BuildConfig;
 
 import static org.kiwix.kiwixmobile.downloader.ChunkUtils.deleteAllParts;
 import static org.kiwix.kiwixmobile.utils.Constants.TAG_KIWIX;
 
 public class FileUtils {
+  private static final String TAG = "FileUtils";
```
Comment (julianharty) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.java]: My suggestion is for ALL TAGs to at least start with the word kiwix as this makes them easier to find when filtering and/or reviewing Android adb logs.

Comment (abdulwd) [reply app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.java]: Done.

**Comment by macgills** [comment]
@kelson42 do you want me to tag and release now?

**Comment by julianharty** [comment]
@kelson42 @macgills Once you make a release available in Google Play I'll spend some time testing it. I'm available to do so for a couple of hours now, else it'll be tomorrow (Tuesday).

**Comment by macgills** [comment]
I started the release process, it will be in alpha in less than an hour hopefully

**Comment by kelson42** [comment]
@macgills Yes please carry on with the release process

**Comment by julianharty** [comment]
I can see you've increased the rollout to 50%. I've tested it on 2 devices so far (one on Monday and a second today) and so far it's behaving well.

**Comment by macgills** [comment]
75% tomorrow and 100% the day after. Really the change is so small I couldn't think there has been a regression and we could have pushed it 100% instantly but it is a good habit to have

**Pull request merged**